[Misc] AMD Build Improvements

842974287 opened the pull request:
Some changes to support Meta internal AMD build. Internally we can't directly use sys.executable, so this PR makes it's possible to modify the subprocess args.

houseroad reviewed (approved):
Looks good to me.

houseroad reviewed (commented):
lint?

houseroad reviewed (commented):
update the title to something like "[AMD] Improve build and packing for AMD build"?

houseroad commented on vllm/model_executor/models/registry.py:
```diff
@@ -201,6 +201,10 @@
     **_FALLBACK_MODEL,
 }
 
+_SUBPROCESS_COMMAND = [
```
this is needed since we would like to override the subprocess command.

We may leverage par format to packing things together, so may need to customize subprocess.run parameters.

houseroad replied:
maybe we should put some comments here.

842974287 replied:
Added comments

842974287 committed:
Changes done internally at Facebook

61d4518a05584a98a79c78cd1ec97e1e2e42df74 Lucia (Lu) Fang <fanglu@meta.com> [vllm][integration] multi-host  TP + PP support in fb path
9ede353980e9181ffe8081cc0f88a93db3824e0e Lu Fang <lufang@meta.com> Move the scripts to vllm scripts folder
590a9e49b79dcef10b23eba0b0fecef447913969 Lu Fang <lufang@meta.com> Clean up github_sync in vllm
935d02cae0dd8b3477a8783dec7902e74ec59972 Lu Fang <lufang@meta.com> Update the OSS script to install pip setuptools_scm package
9d5af8791b3a270e0cc893b1c7661f8376b318e6 Lu Fang <lufang@fb.com> Add AMD build for vllm
7a4b78620798e57ea847c8a57b679327d80789fa dsy842974287 <dsy842974287@fb.com> formatting
acd39af710e710e2c1ad2203fa2b216159641006 dsy842974287 <dsy842974287@fb.com> clean up for PR
33aac23dee6af99fafbecdb63d0c5118295eab80 dsy842974287 <dsy842974287@fb.com> aaa

Signed-off-by: Shiyan Deng <dsy842974287@meta.com>

hongxiayang commented:
LGTM. As long as it can pass the CI/AMD tests, it is good to go.

houseroad commented:
The failed CI seems transient issue, wondering if we have a better way to re-trigger it besides close and reopen the PR, cc: @simon-mo

mgoin reviewed (approved):
I retried the checks, let's see if we can pass

houseroad commented:
AssertionError: The processed dummy data has a total of {'image': 1225, 'video': 4900} placeholder tokens, which is not the expected {'image': 1225, 'video': 3675} token

Multi-modal test failure should be unrelated.

842974287 commented:
Looks like the tests are still failing but the errors don't feel related. 

I saw
```
huggingface_hub.errors.HFValidationError: Repo id must be in the form 'repo_name' or 'namespace/repo_name': '/data/'. Use `repo_type` argument if needed.
...
TypeError: not all arguments converted during string formatting
...
Error: INSTALLATION FAILED: client rate limiter Wait returned an error: context deadline exceeded
helm.go:84: [debug] client rate limiter Wait returned an error: context deadline exceeded
INSTALLATION FAILED
```
in `lint-and-deploy` test.


```
[2025-02-11T19:27:32Z] E       AssertionError: Expected vLLM to gracefully shutdown in <60s if there is an error in the startup.
--
  | [2025-02-11T19:27:32Z] E       assert (9952.833083041 - 9887.64305301) < 60
  | [2025-02-11T19:27:32Z]
  | [2025-02-11T19:27:32Z] mq_llm_engine/test_error_handling.py:254: AssertionError
```
in `Async Engine, Inputs, Utils, Worker Test`.


```
AssertionError: The processed dummy data has a total of {'image': 1225, 'video': 4900} placeholder tokens, which is not the expected {'image': 1225, 'video': 3675} tokens.
```
in `LoRA Test-1`.

```
RuntimeError: Server exited unexpectedly.
```
in both `Quantization Test` and `OpenAI-Compatible Tool Use`.


Wondering if there's any suggestion! cc: @mgoin, @simon-mo
